tests/platform/cloud/AcceptanceTests_oauth2_addPrefs.js

```diff
@@ -0,0 +1,266 @@
+                "gpii_firstDiscovery_language": "en-US"
+            }
+        }
+    }
+};
+
+gpii.tests.cloud.oauth2.addPrefs.transformedPrefs = {
+    "contexts": {
+        "gpii-default": {
+            "name": "Default preferences",
+            "preferences": {
+                "http://registry.gpii.net/common/language": "en-US"
+            }
+        }
+    }
+};
+
+gpii.tests.cloud.oauth2.addPrefs.mainSequence = [
+    { // 0
+        funcName: "gpii.test.cloudBased.oauth2.sendAccessTokenRequestInClientCredentials",
+        args: ["{accessTokenRequest}", "{testCaseHolder}.options"]
+    },
+    { // 1
+        event: "{accessTokenRequest}.events.onComplete",
+        listener: "gpii.test.cloudBased.oauth2.verifyAccessTokenInClientCredentialsResponse",
+        args: ["{arguments}.0", "{accessTokenRequest}"]
+    },
+    { // 2
+        funcName: "gpii.test.cloudBased.oauth2.sendAddPrefsRequest",
+        args: ["{addPrefsRequest}", gpii.tests.cloud.oauth2.addPrefs.prefsData, true, "{accessTokenRequest}.accessToken", "firstDiscovery"]
+    },
+    { // 3
+        event: "{addPrefsRequest}.events.onComplete",
+        listener: "gpii.test.cloudBased.oauth2.verifyAddPrefsResponse",
+        args: ["{arguments}.0", "{addPrefsRequest}", gpii.tests.cloud.oauth2.addPrefs.prefsData]
+    },
+    { // 4
+        funcName: "gpii.test.cloudBased.oauth2.sendGetPrefsRequest",
+        args: ["{getPrefsRequest}", "{addPrefsRequest}", "firstDiscovery"]
+    },
+    { // 5
+        event: "{getPrefsRequest}.events.onComplete",
+        listener: "gpii.test.cloudBased.oauth2.verifyGetPrefsResponse",
+        args: ["{arguments}.0", "{getPrefsRequest}", gpii.tests.cloud.oauth2.addPrefs.prefsData]
+    },
+    { // 6
+        funcName: "gpii.test.cloudBased.oauth2.sendGetPrefsRequest",
+        args: ["{getPrefsRequest}", "{addPrefsRequest}"]
+    },
+    { // 7
+        event: "{getPrefsRequest}.events.onComplete",
+        listener: "gpii.test.cloudBased.oauth2.verifyGetPrefsResponse",
+        args: ["{arguments}.0", "{getPrefsRequest}", gpii.tests.cloud.oauth2.addPrefs.transformedPrefs]
+    }
+];
+
+// To test invalid /add-preferences requests
+gpii.tests.cloud.oauth2.addPrefs.addPrefsSequence = [
+    { // 0
+        funcName: "gpii.test.cloudBased.oauth2.sendAddPrefsRequest",
+        args: ["{addPrefsRequest}", gpii.tests.cloud.oauth2.addPrefs.prefsData, true, "{testCaseHolder}.options.accessToken"]
+    },
+    { // 1
+        event: "{addPrefsRequest}.events.onComplete",
+        listener: "gpii.test.verifyStatusCodeResponse",
+        args: ["{arguments}.0", "{addPrefsRequest}", "{testCaseHolder}.options.expectedStatusCode"]
+    }
+];
+
+gpii.tests.cloud.oauth2.addPrefs.testDefs = [
+    {
+        name: "Acceptance test for adding preferences - a successful entire work flow",
+        scope: "add_preferences",
+        client_id: "client_first_discovery",
+        client_secret: "client_secret_firstDiscovery"
+    },
```

Comment:
It would be good, I think, to collect the testDefs together with the disruptions. Something along these lines:

```
gpii.tests.clous.oauth2.addPrefs.disruptedTests = [
    {
        testDef: {
            ...
        },
        disruptions: ...
    }
];

fluid.each(gpii.tests.clous.oauth2.addPrefs.disruptedTests, function (test) {
    gpii.test.cloudBased.oauth2.bootstrapDisruptedTest(test.testDef, {}, test.disruptions, __dirname);
});
```